thirdparty/djvulibre/kobo-always-utf-8.patch

```diff
@@ -0,0 +1,35 @@
+diff --git a/libdjvu/ByteStream.cpp b/libdjvu/ByteStream.cpp
+index 840885d..4162346 100644
+--- a/libdjvu/ByteStream.cpp
++++ b/libdjvu/ByteStream.cpp
+@@ -671,22 +671,23 @@ urlfopen(const GURL &url,const char mode[])
+        }
+   }
+   return retval?retval:fopen((const char *)url.NativeFilename(),mode);
+-#elif defined(__APPLE__)
+-  return fopen((const char *)url.UTF8Filename(),mode);
+-#else
+-  return fopen((const char *)url.NativeFilename(),mode);
+ #endif
++//#elif defined(__APPLE__)
++  return fopen((const char *)url.UTF8Filename(),mode);
++//#else
```

Comment:
@poire-z Patch better so that it's obvious what happened in the patched source or better as a clean removal?

Reply:
It was a bit hard reading `kobo-always-utf-8.patch` without any other context, I had to look at `djvulibre/libdjvu/ByteStream.cpp` to see what was that now-standalone #endif about. But I got it finally.
I would have gone with just adding something like (dunno the correct syntax, but you get the idea):
```
-#elif defined(__APPLE__)
+#elif defined(__APPLE__) or 1
```
or
```
-#elif defined(__APPLE__)
+#elif 1
```
or
```
-#elif defined(__APPLE__)
+#elif defined(__APPLE__) or defined(__KOREADER__) // or any existing var we already define
```
so there's less commenting, and it's obvious we want that APPLE behaviour of UTF8 here.

Reply:
That's what I'm saying though. I think it's much clearer when you see the lines commented out in context, in `ByteStream.cpp` in this case. It just looks less pretty as a patch. :-) But fine, I'll change it to `|| 1` or `|| defined(KOBO)` or something like that. :-P

Of course I did just what you suggest in my Android patch for djvulibre, but I think there's a poetic quality about those lines.

```c
#ifdef __ANDROID__ 
# undef UNIX 
#endif 
```

https://github.com/koreader/koreader-base/blob/031eb1583a243b3dbe1cec48807af33e9138e31a/thirdparty/djvulibre/android.patch#L1-L31